src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -91,13 +101,13 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
             m_properties.limits.shaderSubgroupQuad = vulkan11Properties.subgroupSupportedOperations & VK_SUBGROUP_FEATURE_QUAD_BIT;
             m_properties.limits.shaderSubgroupQuadAllStages = vulkan11Properties.subgroupQuadOperationsInAllStages;
             
-            /* Vulkan Core 1.2 */
+            /* Vulkan 1.2 Core  */
             m_properties.driverID = static_cast<E_DRIVER_ID>(vulkan12Properties.driverID);
             memcpy(m_properties.driverName, vulkan12Properties.driverName, VK_MAX_DRIVER_NAME_SIZE);
             memcpy(m_properties.driverInfo, vulkan12Properties.driverInfo, VK_MAX_DRIVER_INFO_SIZE);
             m_properties.conformanceVersion = vulkan12Properties.conformanceVersion;
 
-            /* Vulkan Core 1.3 */
+            /* Vulkan 1.3 Core  */
```

Comment:
in order for this to work, your API-version would need to be at least that high, the API version can still be lower (like 1.1) and the same functionality provided by extensions.

_Vulkan will not process the `VkPhysicalDeviceVulkan12Properties` or `VkPhysicalDeviceVulkan13Properties` structs if the API version isn't there!_

So I'd suggest instead of having `VkPhysicalDeviceVulkan12Properties` and `VkPhysicalDeviceVulkan13Properties` in your `pNext` chain, have those instead:

### 1.2

```
VkPhysicalDeviceDriverPropertiesKHR
VkPhysicalDeviceFloatControlsPropertiesKHR
VkPhysicalDeviceDescriptorIndexingPropertiesEXT
VkPhysicalDeviceDepthStencilResolvePropertiesKHR
VkPhysicalDeviceSamplerFilterMinmaxPropertiesEXT
VkPhysicalDeviceTimelineSemaphorePropertiesKHR
``` 

if you really want to know ` framebufferIntegerColorSampleCounts` (not provided by any other `pNext` extension struct) then you can shove the `VkPhysicalDeviceVulkan12Properties` into the pNext as well, but you need to check 1.2 API version before you assign its value to anything, and also only use the `VkPhysicalDeviceVulkan12Properties` struct for checking that one value.
 

### 1.3

```
VkPhysicalDeviceSubgroupSizeControlPropertiesEXT
VkPhysicalDeviceInlineUniformBlockPropertiesEXT
VkPhysicalDeviceShaderIntegerDotProductPropertiesKHR
VkPhysicalDeviceTexelBufferAlignmentPropertiesEXT
VkPhysicalDeviceMaintenance4PropertiesKHR
```

Same story as with the 1.2 property struct, but regarding `maxInlineUniformTotalSize`, if you wish to know it, put it in `pNext` read its value only for that and nothing else, and check for API version 1.3 before you read it.

## Conclusion 

You need not worry about reporting discrepancies as Vulkan spec states:

> The members of VkPhysicalDeviceVulkan12Properties must have the same values as the corresponding members of [VkPhysicalDeviceDriverProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceDriverProperties.html), [VkPhysicalDeviceFloatControlsProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceFloatControlsProperties.html), [VkPhysicalDeviceDescriptorIndexingProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceDescriptorIndexingProperties.html), [VkPhysicalDeviceDepthStencilResolveProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceDepthStencilResolveProperties.html), [VkPhysicalDeviceSamplerFilterMinmaxProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceSamplerFilterMinmaxProperties.html), and [VkPhysicalDeviceTimelineSemaphoreProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceTimelineSemaphoreProperties.html).

and

> The members of VkPhysicalDeviceVulkan13Properties must have the same values as the corresponding members of [VkPhysicalDeviceInlineUniformBlockProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceInlineUniformBlockProperties.html) and [VkPhysicalDeviceSubgroupSizeControlProperties](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceSubgroupSizeControlProperties.html).

Reply:
Also when we go further to get extensions strings based on "enabled features", we should check for API version and if it wasn't high enough we should add the corresponding extension providing those features.
for example when user wants to enable descriptor indexing stuff and descriptorIndexing is either in `core 1.2` or `VK_EXT_descriptor_indexing` so if  API version is 1.1 we add `VK_EXT_descriptor_indexing` to our extensions string we give to device creation params

Reply:
> Also when we go further to get extensions strings based on "enabled features", we should check for API version and if it wasn't high enough we should add the corresponding extension providing those features. for example when user wants to enable descriptor indexing stuff and descriptorIndexing is either in `core 1.2` or `VK_EXT_descriptor_indexing` so if API version is 1.1 we add `VK_EXT_descriptor_indexing` to our extensions string we give to device creation params

why not always just use the extension?

Is it possible to get a `core 1.2` API implementation which does not advertise `VK_EXT_descriptor_indexing` or will not understand it properly?

Reply:
Old comment, resolved now that we know we don't need those structures

Reply:
Basically I misunderstood some stuff, nevermind

Reply:
ok